### PR TITLE
CIRC-467: add lost item fee policy in circulation rules syntax

### PR DESCRIPTION
**Pull request by VRohach**
## Purpose

JIRA ticket: https://issues.folio.org/browse/CIRC-467
## Approach

1. New symbol 'i' added to the circulation rules editor (lost item fee policy)
2. Mechanism for retrieving "lostItemPolicyId" created.
3. Tests were updated. New tests for 'i' symbol were added. 

This PR **is similar** to the previous PR regarding adding a new letter 'o' (overdue fine policy), which **_was reviewed and merged_**: https://github.com/folio-org/mod-circulation/pull/370
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by VRohach** [comment]
This PR should be merged with CIRCSTORE-171 for avoiding unexpected behavior in the circulation rules editor. PR: https://github.com/folio-org/mod-circulation-storage/pull/219

File: doc/circulationrules.md [hunk 1]
```diff
@@ -5,17 +5,17 @@ patron group and the item's material type, loan type, and location.
...
+    m streaming-subscription: l policy-s r no-requests n no-notices o overdue i lost-item
+        g visitor undergrad: l in-house r no-requests n no-notices o overdue i lost-item
 
 How does this short example work?
 
 The `fallback-policy` line defines a default policy for each of the 4 policy types.
-An `l` indicates a loan policy, `r` a request policy, `n` a notice policy, and `o` 
-an overdue fine policy'.
+An `l` indicates a loan policy, `r` a request policy, `n` a notice policy, `o` 
+an overdue fine policy' and `i` lost item fee policy.
```
Comment (julianladisch) [inline doc/circulationrules.md]:
```suggestion
an overdue fine policy' and `i` a lost item fee policy.
```

File: src/test/java/api/support/builders/LostItemFeePolicyBuilder.java [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+      this.id,
+      this.name,
+      this.description,
+      this.itemAgedLostOverdue,
+      this.patronBilledAfterAgedLost,
+      this.chargeAmountItem,
+      this.lostItemProcessingFee,
+      this.chargeAmountItemPatron,
+      this.chargeAmountItemSystem,
+      lostItemChargeFeeFine,
```
Comment (julianladisch) [inline src/test/java/api/support/builders/LostItemFeePolicyBuilder.java]:
```suggestion
      this.lostItemChargeFeeFine,
```

File: src/test/java/org/folio/circulation/rules/Text2DroolsTest.java [hunk 3]
```diff
@@ -99,6 +99,19 @@ public void headerFallbackPolicy() {
...
+    { "foo",       "foo",           "foo",       SECOND_INSTITUTION_ID,                                                     "lost-item" },
+    { "book",      "regular",       "undergrad", FIRST_INSTITUTION_ID,                            "lost-item-9", "lost-item-1", "lost-item" },
+    { "book",      "special-items", "undergrad", FIRST_INSTITUTION_ID,  "lost-item-6", "lost-item-7", "lost-item-9", "lost-item-1", "lost-item" },
+    { "book",      "special-items", "visitor",   SECOND_INSTITUTION_ID, "lost-item-8", "lost-item-6", "lost-item-7", "lost-item-1", "lost-item" },
+    { "newspaper", "regular",       "undergrad", FIRST_INSTITUTION_ID,                            "lost-item-9", "lost-item-2", "lost-item" },
+    { "newspaper", "special-items", "undergrad", FIRST_INSTITUTION_ID,               "lost-item-7", "lost-item-9", "lost-item-2", "lost-item" },
+    { "newspaper", "special-items", "visitor",   FIRST_INSTITUTION_ID,  "lost-item-8", "lost-item-7", "lost-item-9", "lost-item-2", "lost-item" },
+    { "foo",       "foo",           "foo",       FIRST_INSTITUTION_ID,                                         "lost-item-9", "lost-item" },
```
Comment (julianladisch) [inline src/test/java/org/folio/circulation/rules/Text2DroolsTest.java]: Can you align them for readability?

File: doc/circulationrules.md (hunk 1 above)
Comment (VRohach) [reply doc/circulationrules.md]: Thank you! Done.

File: src/test/java/api/support/builders/LostItemFeePolicyBuilder.java (hunk 2 above)
Comment (VRohach) [reply src/test/java/api/support/builders/LostItemFeePolicyBuilder.java]: Done.

File: src/test/java/org/folio/circulation/rules/Text2DroolsTest.java (hunk 3 above)
Comment (VRohach) [reply src/test/java/org/folio/circulation/rules/Text2DroolsTest.java]: Resolved.

**Pull request merged**